csharp/src/Ice/ConnectionI.cs

```diff
@@ -969,6 +969,12 @@ public override void message(ThreadPoolCurrent current)
                             }
                         }
 
+                        if (_maxDispatches > 0 && _dispatchCount == _maxDispatches)
+                        {
+                            // Only read from the connection if max dispatches isn't reached.
+                            newOp &= ~SocketOperation.Read;
```

Comment:
This removes Read from newOp, right?

Reply:
Yes, I'll move this code to `parseMessage` instead, it should be a little clearer.